libsql/src/hrana/stream.rs

```diff
@@ -350,30 +351,43 @@ where
         self.status = StreamStatus::Closed;
     }
 
-    async fn close(&mut self) -> Result<()> {
-        if self.status == StreamStatus::Closed {
-            return Ok(());
-        }
-        self.send(StreamRequest::Close).await?;
-        self.done();
-        Ok(())
-    }
-
     fn next_sql_id(&mut self) -> SqlId {
         self.sql_id_generator = self.sql_id_generator.wrapping_add(1);
         self.sql_id_generator
     }
 }
 
+#[cfg(feature = "remote")]
+impl<T> Drop for RawStream<T>
+where
+    T: HttpSend,
+{
+    fn drop(&mut self) {
+        if self.status == StreamStatus::Closed {
+            return;
+        }
+        tracing::trace!("closing stream");
+        let req = serde_json::to_string(&ClientMsg {
+            baton: self.baton.clone(),
+            requests: vec![StreamRequest::Close],
+        })
+        .unwrap();
+        self.client
+            .clone()
+            .oneshot(self.pipeline_url.clone(), self.auth_token.clone(), req);
+        self.done();
```

Comment:
related to my comment above I think it makes more sense here to do all the spawn logic, unless I am missing something with wasm that makes this slightly more complicated.